application/controllers/admin/export.php

```diff
@@ -867,32 +867,38 @@ public function exportMultipleSurveys($sSurveys, $sExportType)
             if ($iSurveyID === false) {
                 continue;
             }
-            if (Permission::model()->hasSurveyPermission($iSurveyID, 'responses', 'export')) {
-                $archiveName                    = "";
-                $oSurvey                        = Survey::model()->findByPk($iSurveyID);
-                $aResults[$iSurveyID]['title']  = ellipsize($oSurvey->correct_relation_defaultlanguage->surveyls_title, 30);
-                $aResults[$iSurveyID]['result'] = false;
-
+            $archiveName                    = "";
+            $oSurvey                        = Survey::model()->findByPk($iSurveyID);
+            $aResults[$iSurveyID]['title']  = ellipsize($oSurvey->correct_relation_defaultlanguage->surveyls_title, 30);
+            $aResults[$iSurveyID]['result'] = false;
+            if (Permission::model()->hasSurveyPermission($iSurveyID, 'surveycontent', 'export')) {
                 // Specific to each kind of export
                 switch ($sExportType) {
                     // Export archives for active surveys
                     case 'archive':
-                        if ($oSurvey->isActive) {
-                            $archiveName = $this->_exportarchive($iSurveyID, false);
-
-                            if (is_file($archiveName)) {
-                                $aResults[$iSurveyID]['result'] = true;
-                                $aResults[$iSurveyID]['file']   = $archiveName;
-                                $bArchiveIsEmpty                = false;
-                                $archiveFile                    = $archiveName;
-                                $newArchiveFileFullName         = 'survey_archive_'.$iSurveyID.'.lsa';
-                                $this->_addToZip($zip, $archiveFile, $newArchiveFileFullName);
-                                unlink($archiveFile);
+                        if (
+                            ($oSurvey->hasTokensTable && !Permission::model()->hasSurveyPermission($iSurveyID, 'tokens', 'export'))
+                            || !Permission::model()->hasSurveyPermission($iSurveyID, 'responses', 'export')
+                        ) {
+                            $aResults[$iSurveyID]['error'] = gT("We are sorry but you don't have permissions to do this.");
```

Comment:
I think a single work are better : No permission
![Capture d’écran du 2022-07-01 17-27-31](https://user-images.githubusercontent.com/1439428/176924411-ba7ec640-4f83-4a63-943f-a6a3a98a3615.png)

But i merge it like that (security issue)